### DISCUSSION TITLE
ID Generation for Nodes, Links, and Ports

### DISCUSSION THREAD

**Discussion by Qbijayapdl07** [clientIO/joint#2770] [Q&A]
### Introduction

"When creating nodes, links, or ports in JointJS, the library automatically generates a UUID as the ID for each element. Is it possible to generate the ID in TSID format instead?"

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
You can override the `generateId()` and `generatePortId()` cell methods.
_Note: `generatePortId()` returns `generateId()` in the default implementation._

```ts
dia.Element.define('MyShape', {
  // element attributes
}, {
  // element prototype  
  generateId() {
    return TSID.Factory.getTsid().toLong();
  }
}
```
